pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1104,7 +1104,24 @@ func (c *Cloud) ExternalID(nodeName types.NodeName) (string, error) {
 // InstanceExistsByProviderID returns true if the instance with the given provider id still exists and is running.
 // If false is returned with no error, the instance will be immediately deleted by the cloud controller manager.
 func (c *Cloud) InstanceExistsByProviderID(providerID string) (bool, error) {
-	return false, errors.New("unimplemented")
+	instanceID, err := kubernetesInstanceID(providerID).mapToAWSInstanceID()
+	if err != nil {
+		return false, err
+	}
+
+	request := &ec2.DescribeInstancesInput{
+		InstanceIds: []*string{instanceID.awsString()},
+	}
+
+	instances, err := c.ec2.DescribeInstances(request)
+	if err != nil {
+		return false, err
+	}
+	if len(instances) == 0 {
+		return false, nil
+	}
+
+	return true, nil
```

Comment:
I think you need to check the running state.

Also, this duplicates logic in `ExternalID`, so I propose one calls into the other.

Reply:
On second thoughts sharing code looks complicated, but the logic shouldn't change IMO